### PR TITLE
feat(catalog-plugin): Add support for setting placeholder resolvers.

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

Adding an additional extension point for the catalog plugin, needed for #17535.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-node/package.json
Comment (freben) [inline plugins/catalog-node/package.json:43]: This is not possible - it makes a circular dependency. The type needs to be deprecated in the backend and introduced in -node instead.

Comment (sennyeya) [reply plugins/catalog-node/package.json]: Updated.

File: plugins/catalog-backend/src/modules/core/PlaceholderProcessor.ts
Comment (freben) [inline plugins/catalog-backend/src/modules/core/PlaceholderProcessor.ts:60]:
```suggestion
 */
```

File: plugins/catalog-node/src/alpha.ts
Comment (freben) [inline plugins/catalog-node/src/alpha.ts:21]: These aren't alpha though. They should all go in the regular types instead.

If you were hitting problems with importing from alpha code to non-alpha code in the same library, what you need to do is to do it via the package name instead of via local paths, as if you were an external plugin. I don't expect the alpha types to change because of this move of things to the node package.

**Review by freben — APPROVED** [review]
Looks good to me.

@benjdlambert @Rugvip @backstage/catalog-maintainers any qualms about this addition to the extension? It's a bit oddly built internally with how it's actually a dedicated processor, whose interface leaks all the way out to the surface of the catalog itself. On the other hand, it's such an intrinsic feature at this point and shipped per default, that it's inseparable from the catalog itself (arguably). So maybe it's fine, and we can regard the processor as just "an implementation detail" anyway, rather than worrying about wanting to separate out the processor as a separate module which you configure with these resolvers.

**Comment by sennyeya** [comment]
@freben Can I :shipit: or are we waiting for additional feedback?

**Comment by benjdlambert** [comment]
@sennyeya think this just needs a rebase to get the build green and we're good

**Pull request merged**
